tests/Functional/CustomTestKernel.php

```diff
@@ -0,0 +1,247 @@
+
+/*
+ * Needed by PluginInteractionTest, so the test uses the cache for geoCoder, and doesn't clear it each time
+ * BUT doesn't use the cache for the service parameters, like it is happening in dev
+ *
+ * warmupDir is redefined because the method initializeContainer() is using it, but it's private in the parent.
+ * the methods using it (reboot() and getKernelParameters() and setAnnotatedClassCache() ) therefore needed to be redeclared, in order
+ * for them to have a correct value in it.
+ */
+class CustomTestKernel extends TestKernel
+{
+    private $warmupDir;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reboot(?string $warmupDir)
+    {
+        $this->shutdown();
+        $this->warmupDir = $warmupDir;
+        $this->boot();
+    }
+
+    /*
+     * Needed, otherwise the used cache is different on each kernel boot, which is a big issue in PluginInteractionTest
+     */
+    public function getCacheDir(): string
+    {
+        return realpath(sys_get_temp_dir()).'/NyholmBundleTest/cachePluginInteractionTest';
+    }
+
+    /**
+     * Returns the kernel parameters.
+     */
+    protected function getKernelParameters(): array
+    {
+        $bundles = [];
+        $bundlesMetadata = [];
+
+        foreach ($this->bundles as $name => $bundle) {
+            $bundles[$name] = \get_class($bundle);
+            $bundlesMetadata[$name] = [
+                'path' => $bundle->getPath(),
+                'namespace' => $bundle->getNamespace(),
+            ];
+        }
+
+        return [
+            'kernel.project_dir' => realpath($this->getProjectDir()) ?: $this->getProjectDir(),
+            'kernel.environment' => $this->environment,
+            'kernel.runtime_environment' => '%env(default:kernel.environment:APP_RUNTIME_ENV)%',
+            'kernel.debug' => $this->debug,
+            'kernel.build_dir' => realpath($buildDir = $this->warmupDir ?: $this->getBuildDir()) ?: $buildDir,
+            'kernel.cache_dir' => realpath($cacheDir = ($this->getCacheDir() === $this->getBuildDir() ? ($this->warmupDir ?: $this->getCacheDir()) : $this->getCacheDir())) ?: $cacheDir,
+            'kernel.logs_dir' => realpath($this->getLogDir()) ?: $this->getLogDir(),
+            'kernel.bundles' => $bundles,
+            'kernel.bundles_metadata' => $bundlesMetadata,
+            'kernel.charset' => $this->getCharset(),
+            'kernel.container_class' => $this->getContainerClass(),
+        ];
+    }
+
+    /**
+     * @internal
+     */
+    public function setAnnotatedClassCache(array $annotatedClasses)
+    {
+        file_put_contents(($this->warmupDir ?: $this->getBuildDir()).'/annotations.map', sprintf('<?php return %s;', var_export($annotatedClasses, true)));
+    }
+
+    /**
+     * Initializes the service container.
+     *
+     * The built version of the service container is used when fresh, otherwise the
+     * container is built.
+     */
+    protected function initializeContainer()
```

Comment:
same question here :)

Reply:
This isn't the same method, even though the modification is tiny (I thought I should comment it, but I didn't :( )
Line 108: I placed a false : "(false && \is_object($this->container = include $cachePath)"
Line 129: I placed a true : "elseif (true || !\is_object($this->container = include $cachePath))".
I know it's not pretty, but I basically wanted the test not to use the cache for the container parameters, but to do it for the geocoder, and didn't find any setting to do that.